src/lib/WalletInfoCollector.ts

```diff
@@ -72,20 +77,33 @@ export default class WalletInfoCollector {
         if (initialAccounts.length > 0) {
             WalletInfoCollector._addAccounts(walletInfo, initialAccounts, undefined);
 
-            if (walletType !== WalletType.LEGACY) {
-                // fetch balances and update again
-                initialAccountsPromise = WalletInfoCollector._getBalances(initialAccounts).then(async (balances) => {
-                    WalletInfoCollector._addAccounts(walletInfo, initialAccounts, balances);
-                    onUpdate(walletInfo, await derivedAccountsPromise);
-                });
-            }
+            // fetch balances and update again
+            // This step could be skipped for WalletType.LEGACY in case there is only one Account to collect info on.
+            // In that case the WalletInfoCollector needs to wait for consensus unnecessarily.
```

Comment:
This is indeed unfavorable. When importing a definite legacy account, one should not have to wait for a consensus.